truffle/src/main/java/org/jruby/truffle/nodes/core/TruffleInteropNodes.java

```diff
@@ -203,21 +193,21 @@ private static String slowPathToString(RubyString identifier) {
     @CoreMethod(names = "write_property", isModuleFunction = true, needsSelf = false, required = 3)
     public abstract static class WritePropertyNode extends CoreMethodArrayArgumentsNode {
 
-        @Child private ForeignObjectAccessNode node;
+        @Child private Node node;
 
         public WritePropertyNode(RubyContext context, SourceSection sourceSection) {
             super(context, sourceSection);
-            this.node = ForeignObjectAccessNode.getAccess(Write.create(Receiver.create(), Argument.create(), Argument.create()));
+            this.node = ForeignAccess.node(ForeignAccess.msgWrite());
```

Comment:
I think as well `createNode` would be clearer if it is a factory method, `.node` could be like a getter, or possibly implies some caching (does it?).
Would an enum or constants make sense for the msgXYZ if they are singleton instances?